main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
> I'm not 100% sure of what is the "correct" behavior regarding naked \r on non-Macintosh OSes. It would be interesting to check how different libcs implement gets() in that regard -- the manpage on my system only talks about \n, so the behavior here was correct, but maybe it's different with \r sequences depending on the OS.

Yeah, this is the only possibly "unclean" thing about this patch because we should handle \r somehow and I'm not sure whether this will be compatible with libc (will have a look).